views/src/main/java/io/micronaut/views/thymeleaf/ThymeleafFactory.java

```diff
@@ -68,6 +68,8 @@ public AbstractConfigurableTemplateResolver templateResolver(ViewsConfiguration
     @Singleton
     public TemplateEngine templateEngine(ITemplateResolver templateResolver) {
         TemplateEngine engine = new TemplateEngine();
+        engine.setEngineContextFactory(new EngineContextFactory());
+        engine.setLinkBuilder(new LinkBuilder());
```

Comment:
These items should be injected instead of directly instantiated

Reply:
Can't think of a good use outside TemplateEngine configuring.

Reply:
@gasches Someone may want to provide their own engine context factory and override your implementation. With the code as it is that is not possible without overriding the entire template engine. Same goes for the link builder